src/exasolvs/AdapterProperties.lua

```diff
@@ -0,0 +1,164 @@
+local text = require("text")
+local exaerror = require("exaerror")
+
+--- This class abstracts access to the user-defined properties of the Virtual Schema.
+-- @type AdapterProperties
+local AdapterProperties = {}
+
+local EXCLUDED_CAPABILITIES_PROPERTY <const> = "EXCLUDED_CAPABILITIES"
+local LOG_LEVEL_PROPERTY <const> = "LOG_LEVEL"
+local DEBUG_ADDRESS_PROPERTY <const> = "DEBUG_ADDRESS"
+local DEFAULT_LOG_PORT <const> = 3000
+
+--- Factory method for adapter properties
+-- @param raw_properties comma-separated list of property names
+-- @return adapter properties object
+function AdapterProperties.create(raw_properties)
+    return AdapterProperties:new({raw_properties = raw_properties})
+end
+
+--- Create a new instance of adapter properties.
+-- @param object pre-initialized object
+-- @return new instance
+function AdapterProperties:new (object)
+    object = object or {raw_properties = {}}
+    self.__index = self
+    setmetatable(object, self)
+    return object
+end
+
+
+--- Get the value of a property.
+-- @param property_name name of the property to get
+-- @return property value
+function AdapterProperties:get(property_name)
+    return self.raw_properties[property_name]
+end
+
+--- Check if the property is set.
+-- @param property_name name of the property to check
+-- @return <code>true</code> if the property is set (i.e. not <code>nil</code>)
+function AdapterProperties:is_property_set(property_name)
+    return self:get(property_name) ~= nil
+end
+
+--- Check if the property has a non-empty value.
+-- @param property_name name of the property to check
+-- @return <code>true</code> if the property has a non-empty value (i.e. not <code>nil</code> or an empty string)
+function AdapterProperties:has_value(property_name)
+    local value = self:get(property_name)
+    return value ~= nil and value ~= ""
+end
+
+--- Check if the property value is empty.
+-- @param property_name name of the property to check
+-- @return <code>true</code> if the property's value is empty (i.e. the property is set to an empty string)
+function AdapterProperties:is_empty(property_name)
+    return self:get(property_name) == ""
+end
+
+function AdapterProperties:_validate_debug_address()
+    if self:has_value(DEBUG_ADDRESS_PROPERTY) then
+        local address = self:get(DEBUG_ADDRESS_PROPERTY)
+        if not string.match(address, "^.-:[0-9]+$") then
+            exaerror.create("F-VSCL-PROP-3", "Expected log address in " .. DEBUG_ADDRESS_PROPERTY
+                    .. " to look like '<ip>|<host>[:<port>]', but got {{address}} instead"
```

Comment:
I prefer to use string.format() to format string rather than using ".." . 
As can be seen from the whole project, you prefer to use "..".

Reply:
In this case in particular, I find it more readable, because you otherwise get stacking string interpolation. And the part that users can't influence is fixed in the message, so that it does not appear as parameter in the error catalog.